packages/langium/src/grammar/langium-grammar.langium

```diff
@@ -50,42 +50,46 @@ GrammarImport:
     'import' path=STRING ';'?;
 
 ParserRule:
-    (entry?='entry' | fragment?='fragment')? RuleNameAndParams (wildcard?='*' | 'returns' type=ReturnType | infers?='infers' type=ReturnType)?
+    (entry?='entry' | fragment?='fragment')?
+    RuleNameAndParams
+    (wildcard?='*' | ('returns' (returnType=[AbstractType:ID] | dataType=PrimitiveType)) | inferredType=InferredType)?
     (definesHiddenTokens?='hidden' '(' (hiddenTokens+=[AbstractRule:ID] (',' hiddenTokens+=[AbstractRule:ID])*)? ')')? ':'
-        alternatives=Alternatives
-    ';';
+    alternatives=Alternatives ';';
+
+InferredType:
+    'infer' name=ID;
```

Comment:
We should keep the syntax the same:

```suggestion
InferredType<imperative>:
    (<imperative> 'infer' | <!imperative> 'infers') name=ID;
```

Reply:
After all the discussions to this topic I don't really recall what was the final outcome.
I thought it was that we use only `infer` to not have `infer` and `infers` which is easy to mix up and difficult to remember

Reply:
We decided on `infer` for actions and `infers` for parser rules. That's how we released Langium 0.3

Reply:
Okay, this doesn't really affects the outstanding work, needs just yet another round of boostraping orgy